cloudify_nsx/network/bgp_neighbour_filter.py

```diff
@@ -98,10 +99,16 @@ def delete(**kwargs):
     # credentials
     client_session = common.nsx_login(kwargs)
 
-    cfy_dlr.del_bgp_neighbour_filter(
-        client_session,
-        resource_id
-    )
+    try:
+        cfy_dlr.del_bgp_neighbour_filter(
+            client_session,
+            resource_id
+        )
+    except Exception as ex:
+        ctx.logger.error("We have issue with remove: %s", str(ex))
```

Comment:
It might be better to approach this as:
ctx.logger.error("Error trying to delete resource: %s" % str(ex))
raise cfy_exc.RecoverableError(
    message='Failed to deleted resource, retrying..."
)

I'm not sure about assigning the retry delay either- that's something that we have generally been doing via the --task-retry-interval argument?

(this applies to the other similar logic below- and maybe it would be worth factoring out all of this logic into the common library- then just call:
attempt_delete(
    session=client_session,
    resource_id=resource_id,
    delete_function=cfy_dlr.del_bgp_neighbour_filter,
)
(or even pass the entire thing in using functools.partial)